ibm-eventstore/src/test/java/akka/stream/alpakka/ibm/eventstore/javadsl/EventStoreSpec.java

```
@@ -0,0 +1,111 @@
+        fields.add(DataTypes.createStructField("someString", DataTypes.StringType, false));
+        fields.add(DataTypes.createStructField("someBoolean", DataTypes.BooleanType, false));
+        fields.add(DataTypes.createStructField("someOtherBoolean", DataTypes.BooleanType, true));
+
+        return new TableSchema(configuration.tableName(), DataTypes.createStructType(fields), pkList, pkList, scala.Option.apply(null));
+    }
+
+    @BeforeClass
+    public static void setup() {
+        final Pair<ActorSystem, Materializer> sysmat = setupMaterializer();
+        system = sysmat.first();
+        materializer = sysmat.second();
+
+        // #configure-endpoint
+        ConfigurationReader.setConnectionEndpoints(configuration.endpoint());
+        // #configure-endpoint
+        EventContext.dropDatabase(configuration.databaseName());
+        EventContext context = EventContext.createDatabase("TESTDB");
+        TableSchema reviewSchema = getTableSchema();
+        context.createTable(reviewSchema);
+    }
+
+    @AfterClass
+    public static void teardown() {
+        ConfigurationReader.setConnectionEndpoints(configuration.endpoint());
+        EventContext.dropDatabase(configuration.databaseName());
+        // #cleanup
+        EventContext.cleanUp();
+        // #cleanup
+        JavaTestKit.shutdownActorSystem(system);
+    }
+
+    @Test
+    public void testInsertingRecordsIntoTable() throws Exception {
+
+        //#insert-rows
+        List<Row> rows = new ArrayList<>();
+        rows.add(RowFactory.create(1, 1, "Hello", true, false));
+        rows.add(RowFactory.create(2, 1, "Hello", true, false));
+        rows.add(RowFactory.create(3, 1, "Hello", true, false));
+
+        Sink<Row, CompletionStage<Done>> sink = EventStoreSink.create(configuration, materializer.executionContext());
+        final CompletionStage<Done> insertionResultFuture = Source.from(rows).runWith(sink, materializer);
+        //#insert-rows
```

Is there any way we could tell from the native client return values that insertion actually happened? If so, that might be a better materialized value then `Done` and it would help make this unit test a little better, because now we don't know that anything actually happened.

---

To obtain the result, the user can use the flow version

---

Does the Sink fail with error if it cannot insert? Or does it silently fail? Because that would be very bad.

---

(Also, that should be tested, that it does fail)